tests/helics/shared_library/test-message-federate.cpp

```diff
@@ -110,8 +110,14 @@ TEST_P (mfed_simple_type_tests, send_receive)
     auto M = helicsEndpointGetMessage (epid2);
     // ASSERT_TRUE (M);
     ASSERT_EQ (M.length, 500);
-
-    EXPECT_EQ (M.data[245], 'a');
+	if (M.data != nullptr)
```

Comment:
Is there a reason for doing the check this way instead of using `ASSERT_NE(M.data, nullptr)` like further down in the mfed_type_tests::send_receive_2fed test?

Reply:
not particularly other than mild curiosity which ones klokwork actually recognizes to clear up a warning.  pretty sure M.data can't be NULL if length!=0 but for some reason klokwork doesn't think that is sufficient enough and I am not sure it recognizes the check in ASSERT_EQ.